src/components/Details/VmDetails/VmDetails.js

```diff
@@ -0,0 +1,252 @@
+        return <props.ResourceLink kind={kind} name={name} namespace={namespace} title={uid} />;
+      }
+    }
+  }
+  return DASHES;
+};
+
+const NodeResourceLink = props => {
+  const { loaded, vm, NodeLink, resources } = props;
+  let nodeName = null;
+
+  if (loaded) {
+    const pod = getPod(vm, resources, VIRT_LAUNCHER_POD_PREFIX);
+    nodeName = pod ? pod.spec.nodeName : null;
+  }
+
+  return nodeName ? <NodeLink name={nodeName} /> : DASHES;
+};
+
+NodeResourceLink.propTypes = {
+  loaded: PropTypes.bool.isRequired,
+  vm: PropTypes.object.isRequired,
+  NodeLink: PropTypes.func.isRequired,
+  resources: PropTypes.array.isRequired,
+};
+
+const OnOffReporter = props => {
+  const statusText = props.on ? 'On' : 'Off';
+  const statusIcon = props.on ? 'pficon pficon-on' : 'pficon pficon-off';
+  const statusColor = props.on ? '#3f9c35' : '#a30000';
+
+  return (
+    <span>
+      <span style={{ color: `${statusColor}` }} className={statusIcon} /> {statusText}
+    </span>
+  );
+};
+
+OnOffReporter.propTypes = {
+  on: PropTypes.bool.isRequired,
+};
+
+const VmStatusReporter = props => {
+  const vmIsRunning = props.vm.spec.running;
+  const statusText = vmIsRunning ? 'on' : 'off';
+  const statusIcon = vmIsRunning ? 'pficon pficon-on' : 'pficon pficon-off';
+  const statusColor = vmIsRunning ? '#3f9c35' : '#a30000';
+
+  return (
+    <span style={{ color: `${statusColor}` }}>
+      <span className={statusIcon} /> {props.vm.metadata.name} is {statusText}
+    </span>
+  );
+};
+
+VmStatusReporter.propTypes = {
+  vm: PropTypes.object.isRequired,
+};
+
+export const VmDetails = props => {
+  const { loaded, NodeLink, resources, ResourceLink, vm } = props;
+  const pod = getPod(vm, resources, VIRT_LAUNCHER_POD_PREFIX);
+
+  // TODO Fix FieldLevelHelp content for Status
+  return (
+    <Fragment>
+      <div className="co-m-pane__body">
+        <h1 className="co-m-pane__heading">Overview</h1>
+        <div className="co-m-pane__body">
+          <div className="row">
+            {/* Description column */}
+            <div className="col-sm-4">
+              <div>
+                <VmStatusReporter vm={vm} />
+              </div>
+              <dd>{get(vm, 'metadata.annotations.description', DASHES)}</dd>
+            </div>
+
+            {/* Details columns */}
+            <div className="col-sm-8">
+              <div className="row">
+                {/* Details column 1 */}
+                <div className="col-sm-4">
+                  <dl>
+                    <dt>
+                      Status
+                      <FieldLevelHelp content={"This is the VM's status"} />
+                    </dt>
+                    <dd>
+                      <StateColumn vm={vm} loaded={props.loaded} resources={props.resources} />
+                    </dd>
+
+                    <dt>Operating System</dt>
+                    <dd>{get(vm, 'metadata.annotations["os.template.cnv.io"]', DASHES)}</dd>
```

Comment:
can you extract this to https://github.com/kubevirt/web-ui-components/blob/master/src/utils/selectors.js ? Same for Workload Profile and Templates. 

I also moved the OS/Flavor/Workload from annotations to labels just today. So its under vm/metadata.labels not vm/metadata.annotations

Reply:
Can you please also show the OS icon? In common templates it is defined as `iconClass` which points to [standard openshift icons](https://rawgit.com/openshift/openshift-logos-icon/master/demo.html) defined [here](https://github.com/kubevirt/web-ui/blob/master/frontend/public/components/catalog/catalog-item-icon.tsx).
Pleas note this icon might not be defined in the template so you can not rely on it 100%, but in all standard templates it should be there.